### ISSUE TITLE
moz-extension speech window textarea isn't rewind friendly

### ISSUE THREAD

**Issue by pal1000** [ettoolong/TTSFox#19]
At the moment the speech window textarea isn't rewind friendly at all. 

Consider this scenario: I selected a large chunk of text, often large enough to generate scroll bars to the textarea element; this chunk of text is comprised of multiple paragraphs; let's assume external noise disturbs me and I want to rewind to the beginning of the current paragraph.  This is easier said than done. If I pause the reading task, there is no cursor to use "Start from cursor position". If I stop it, the  textarea is scrolled if it is the case so that the cursor lands either at the begging or at the end of the entire textarea.

There are multiple ways to deal with this:
- implement rewind to the beginning of the current paragraph;
- when stop reading move the cursor at the beginning of the current word or paragraph instead of the end of the entire chunk of text.

**Comment by ettoolong** [comment]
Good, idea.

> rewind to the beginning of the current paragraph

I will implement this.

- After start reading, gray out text that out of your selection range.
- After pause reading:
  - display a cursor to show current position( resume reading will start from the cursor position )
  - add some buttons: **jump to previous paragraph** and **jump to next paragraph** (when mouse over these buttons, display underline to show paragraph.)
  - add some buttons: **jump to previous sentence** and **jump to next sentence** (when mouse over these buttons, display underline to show sentence.), I am not sure if this is possible, because different languages are using different symbol between sentences.

Have any other idea about this plan?

> when stop reading move the cursor at the beginning of the current word or paragraph instead of the end of the entire chunk of text

I will not implement this. Move the cursor will cancel the textarea selection text. You need re-selection them. I think keeping the cursor position(or selection range) is better.

**Comment by ettoolong** [comment]
I add 4 new buttons(previous paragraph/previous sentence/next sentence/next paragraph).
![2017-12-12 8 12 07](https://user-images.githubusercontent.com/909570/33893989-c415b150-df97-11e7-9b96-5c3b81a99196.png)

After few days of trial, I remove these buttons.
I found some problems:

- when mouse over these buttons, highlight a sentence is not intuitive enough.
- user click between these buttons to move cursor make the UI more complicated.

Finally, I changed my design:
After pause reading:

- display a cursor to show current position( resume reading will start from the cursor position )
- click on sentence will move cursor to the start of sentence.
- shift + click on sentence to read out it ( cursor would keep on original position ).
